### PR TITLE
The easter refactor

### PR DESCRIPTION
**Pull request by MightyLoggor**
Self explanatory

### CODE REVIEW

File: librarycard.py [hunk 1]
```diff
@@ -926,6 +927,31 @@ def formatBookItemList(items):
...
 
+async def easter_egg(message: discord.message):
+  chance = os.getenv('EASTER_EGG_CHANCE')
+  emoji_list = os.getenv('EASTER_EGG_EMOJI_LIST').split(',')
+  rng = random.random() * 100
+  if rng <= float(chance):
+    random_emoji = emoji_list[random.randint(0, emoji_list.__len__()-1)]
+    await message.add_reaction(random_emoji)
+
+async def royalroad_embbed(message: discord.message):
```
Comment (Grissess) [inline librarycard.py]: typo: `embed`

File: librarycard.py [hunk 2]
```diff
@@ -926,6 +927,31 @@ def formatBookItemList(items):
...
+  if message.content.startswith(("https://www.royalroad.com/fiction/", "https://royalroad.com/fiction/")) :
+        book_url = message.content.split()[0]
+        embed = await getRoyalRoadBook("/".join(book_url.split('/')[:6])) #fixes the url format
+        if embed:
+            await message.channel.send(embed=embed, reference=message.to_reference())
+            await message.edit(suppress = True)
+
+async def goodreads_embbed(message: discord.message):
```
Comment (Grissess) [inline librarycard.py]: idem

File: librarycard.py [hunk 3]
```diff
@@ -936,18 +962,8 @@ async def on_message(message: discord.message):
...
-            await message.edit(suppress = True)
-    
-    if message.content.startswith(("https://www.royalroad.com/fiction/", "https://royalroad.com/fiction/")) :
-        book_url = message.content.split()[0]
-        embed = await getRoyalRoadBook("/".join(book_url.split('/')[:6])) #fixes the url format
-        if embed:
-            await message.channel.send(embed=embed, reference=message.to_reference())
-            await message.edit(suppress = True)
+    await goodreads_embbed(message)
+    await royalroad_embbed(message)
```
Comment (Grissess) [inline librarycard.py]: and here as well :)

on the upside, refactor is a good idea :)

**Review by Grissess — APPROVED** [review]
Typos can be fixed later; if this needs to ship, it's fine as is :P

**Review by Grissess — APPROVED** [review]
Yep, that's all I had; code smell's fine--assuming adequate testing, lgtm :P

**Comment by mdnwvn** [comment]
Final Squash.

**Pull request merged**